### PR TITLE
Azure Monitor: getAzureMonitorRoutes as a single source of the routes

### PR DESCRIPTION
**Pull request by kostrse**
This PR does preparation work for migration of Azure Monitor datasource to common Azure credentials format (#60782) and also for support of custom Azure clouds (#61124).

First, it removes the current prototype implementation of custom routes, and also makes the `getAzureMonitorRoutes` (moved to `routes.go`) to be the single source of routes for the Azure Monitor datasource.

Custom clouds support will be implemented in `getAzureMonitorRoutes` transparently for the rest of the datasource logic.

Azure Portal URL also moved under routes.

### CODE REVIEW

File: pkg/tsdb/azuremonitor/routes.go [hunk 1]
```diff
@@ -49,6 +53,18 @@ var azUSGovLogAnalytics = types.AzRoute{
 	Headers: map[string]string{"x-ms-app": "Grafana", "Cache-Control": "public, max-age=60"},
 }
 
+var azPortal = types.AzRoute{
+	URL: "https://management.azure.com",
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/routes.go:57]: I believe all three of these portal URLs are incorrect. Should they not be:
```
"https://portal.azure.com"
"https://portal.azure.cn"
"https://portal.azure.us"
```

**Review by aangelisc — CHANGES_REQUESTED** [review]
Nice work @kostrse. Some minor comments. Additionally for now I think it'd be prudent for us to leave the current custom routes implementation in. Removing it will break integration tests and will also break the feature if any users are making use of it, wdyt?

File: pkg/tsdb/azuremonitor/routes.go (hunk 1 above)
Comment (kostrse) [reply pkg/tsdb/azuremonitor/routes.go:57]: You are right, I overlooked it.

**Comment by kostrse** [comment]
I disabled the integration test for now. One support for custom Azure clouds implemented (#61124), integration tests will be able to configure mock endpoints via them. It will be applicable for all Azure datasources.

**Comment by aangelisc** [comment]
@kostrse can we close this PR for now and revisit this in future?

**Comment by aangelisc** [comment]
Closing as stale.

**Pull request closed without merging**